src/view/com/profile/FollowButton.tsx

```diff
@@ -20,28 +21,59 @@ export const FollowButton = observer(function FollowButtonImpl({
   onToggleFollow?: (v: boolean) => void
   labelStyle?: StyleProp<TextStyle>
 }) {
-  const {state, following, toggle} = useFollowProfile(profile)
+  const followMutation = useProfileFollowMutation()
+  const unfollowMutation = useProfileUnfollowMutation()
 
-  const onPress = React.useCallback(async () => {
+  const onPressFollow = async () => {
+    if (profile.viewer?.following) {
+      return
+    }
     try {
-      const {following} = await toggle()
-      onToggleFollow?.(following)
+      await followMutation.mutateAsync({did: profile.did})
+      onToggleFollow?.(false)
     } catch (e: any) {
-      Toast.show('An issue occurred, please try again.')
+      Toast.show(`An issue occurred, please try again.`)
     }
-  }, [toggle, onToggleFollow])
+  }
 
-  if (state === FollowState.Unknown) {
+  const onPressUnfollow = async () => {
+    if (!profile.viewer?.following) {
+      return
+    }
```

Comment:
There's a split second where `profile.viewer.following` can be set to `pending`, and this fn should return early in that case too. Rn we're reusing this prop that's typically either `undefined | URI`, so that we know a follow request is in-flight.

Reply:
So, this is based on the pattern in `ProfileHeader` so I'd rather align with that. If it's wrong we should fix it in both places — and probably many more places because `ProfileHeader` uses a similar pattern for a bunch of other handlers. So I'd keep it out of scope of this PR.

Reply:
Added to tracker